src/main/java/cn/nukkit/item/Item.java

```diff
@@ -408,6 +418,26 @@ private static void initCreativeItems() {
                 MainLogger.getLogger().logException(e);
             }
         }
+
+        // New creative items mapping
+        registerCreativeItemsNew(ProtocolInfo.v1_17_0, ProtocolInfo.CURRENT_PROTOCOL, creative440);
+    }
+
+    private static void registerCreativeItemsNew(int protocolId, int blockPaletteProtocol, List<Item> creativeItems) {
+        JsonArray itemsArray;
+        try (InputStream stream = Server.class.getClassLoader().getResourceAsStream("creative_items_" + protocolId + ".json")) {
+            itemsArray = JsonParser.parseReader(new InputStreamReader(stream, StandardCharsets.UTF_8)).getAsJsonObject().getAsJsonArray("items");
+        } catch (Exception e) {
+            throw new AssertionError("Error loading required block states!");
+        }
+
+        for (JsonElement element : itemsArray) {
+            Item item = RuntimeItems.getMapping(protocolId).parseCreativeItem(element.getAsJsonObject(), true, blockPaletteProtocol);
+            if (item != null && !item.getName().equals(UNKNOWN_STR)) {
```

Comment:
This is not going to work properly because there are items that have no own class and therefore won't return correct name

Reply:
I have already verified this. Works fine on official nukkit and in case name does not match, we should solve it by implementing it properly.

Reply:
This causes certain items to be missing in creative inventory

Reply:
Can you write me out the items whivh are affected?